src/weights.jl

```diff
@@ -53,9 +57,34 @@ Base.getindex(wv::W, ::Colon) where {W <: AbstractWeights} = W(copy(wv.values),
     isfinite(sum) || throw(ArgumentError("weights cannot contain Inf or NaN values"))
     wv.values[i] = v
     wv.sum = sum
+    wv.positive = missing
```

Comment:
I guess it would be more efficient to use
```suggestion
    wv.positive = wv.positive === missing ? missing : wv.positive && v >= zero(v)
```
here?

(Again, IMO it's a bit strange to call it `positive` since it rather checks non-negativity.)

Reply:
Indeed, good trick!

Reply:
Actually that doesn't work, as when `positive == false` and `v` is positive, we don't know whether `v` has just replaced the last negative value. So we have to store `missing` rather than `false`. I've pushed something a bit more involved.

Reply:
Ah good catch!